kardinal-cli/deployment/gateway.go

```diff
@@ -66,81 +65,67 @@ func findAvailablePortInRange(host string, portsInUse *[]int) (int, error) {
 	return port, nil
 }
 
-func StartGateway(ctx context.Context, hostFlowIdMap map[string]string) error {
-	client, err := kubernetes_pack.CreateKubernetesClient()
+func StartGateway(hostFlowIdMap []api_types.IngressAccessEntry) error {
+	k8sConfig, err := cli_k8s.GetConfig()
 	if err != nil {
 		return fmt.Errorf("an error occurred while creating a kubernetes client:\n %v", err)
 	}
-
-	// TODO move these values to a shared library between Kardinal Manager, Kontrol and Kardinal CLI
-	kardinalLabelKey := "kardinal.dev"
-	enabledKardinal := "enabled"
-
-	namespaceLabels := map[string]string{
-		kardinalLabelKey: enabledKardinal,
-	}
-
-	namespaceList, err := client.GetNamespacesByLabels(ctx, namespaceLabels)
+	client, err := cli_k8s.CreateKubernetesClient(k8sConfig)
 	if err != nil {
-		return fmt.Errorf("failed to list namespaces from Kubernetes: %v", err)
-	}
-	if len(namespaceList.Items) == 0 {
-		return fmt.Errorf("cannot start gateway because no Kardinal namespace was found")
+		return fmt.Errorf("an error occurred while creating a kubernetes client:\n %v", err)
 	}
-	if len(namespaceList.Items) > 1 {
-		return fmt.Errorf("cannot start gateway because more than one Kardinal namespace was found")
+	gatewayClient, err := cli_k8s.CreateGatewayApiClient(k8sConfig)
+	if err != nil {
+		return fmt.Errorf("an error occurred while creating a kubernetes client:\n %v", err)
 	}
-	baselineNamespace := namespaceList.Items[0]
 
-	for host, flowId := range hostFlowIdMap {
+	servers := make([]*http.Server, 0)
+	ports := make([]int, 0)
+	stopChan := make(chan struct{}, 1)
+
+	for entryIx, entry := range hostFlowIdMap {
+
+		localPort := int32(localPortStartRange + entryIx)
+		host := entry.Hostname
+
 		logrus.Printf("Starting gateway for host: %s", host)
 
-		// Check for pods in the baseline namespace
-		err = assertBaselineNamespaceReady(client.GetClientSet(), flowId, baselineNamespace.Name)
+		err = assertBaselineNamespaceReady(client.GetClientSet(), entry.FlowId, entry.FlowNamespace)
 		if err != nil {
 			return fmt.Errorf("failed to assert that baseline namespace is ready: %v", err)
 		}
 
-		// Check for the Envoy filter before proceeding
-		err = checkGatewayEnvoyFilter(client.GetClientSet(), host)
+		// Find a pod for the service
+		pod, port, namespace, err := findPodForService(client.GetClientSet(), gatewayClient, entry)
 		if err != nil {
-			return err
+			// return fmt.Errorf("failed to find pod for service: %v", err)
```

Comment:
you can remove this commented line.
I think you can handle the no pod found on all cycles case, maybe after the `for` iteration because otherwise the command will get blocked with non-forwarding

Reply:
make sense 👍